src/language.rs

```diff
@@ -30,161 +30,148 @@ pub enum SupportedLanguageName {
-            name_for_ignore_select: "treesitterquery",
-        },
-        SupportedLanguage {
-            language: tree_sitter_json::language(),
-            name: SupportedLanguageName::Json,
-            name_for_ignore_select: "json",
-        },
-        SupportedLanguage {
-            language: tree_sitter_css::language(),
-            name: SupportedLanguageName::Css,
-            name_for_ignore_select: "css",
-        },
-        SupportedLanguage {
-            language: tree_sitter_lua::language(),
-            name: SupportedLanguageName::Lua,
-            name_for_ignore_select: "lua",
-        },
+pub static ALL_SUPPORTED_LANGUAGES: BySupportedLanguage<SupportedLanguage> = {
+    use SupportedLanguage::*;
+    [
+        Rust,
+        Typescript,
+        Javascript,
+        Swift,
+        ObjectiveC,
+        Toml,
+        Python,
+        Ruby,
+        C,
+        Cpp,
+        Go,
+        Java,
+        CSharp,
+        Kotlin,
+        Elisp,
+        Elm,
+        Dockerfile,
+        Html,
+        TreeSitterQuery,
+        Json,
+        Css,
+        Lua,
+    ]
+};
+
+static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
+    [
+        tree_sitter_rust::language(),
+        tree_sitter_typescript::language_tsx(),
+        tree_sitter_javascript::language(),
+        tree_sitter_swift::language(),
+        tree_sitter_objc::language(),
+        tree_sitter_toml::language(),
+        tree_sitter_python::language(),
+        tree_sitter_ruby::language(),
+        tree_sitter_c::language(),
+        tree_sitter_cpp::language(),
+        tree_sitter_go::language(),
+        tree_sitter_java::language(),
+        tree_sitter_c_sharp::language(),
+        tree_sitter_kotlin::language(),
+        tree_sitter_elisp::language(),
+        tree_sitter_elm::language(),
+        tree_sitter_dockerfile::language(),
+        tree_sitter_html::language(),
+        tree_sitter_query::language(),
+        tree_sitter_json::language(),
+        tree_sitter_css::language(),
+        tree_sitter_lua::language(),
     ]
 });
 
+#[inline(always)]
+pub fn get_supported_language_language(supported_language: SupportedLanguage) -> Language {
+    SUPPORTED_LANGUAGE_LANGUAGES[supported_language]
+}
+
+pub static SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT: BySupportedLanguage<&'static str> = [
+    "rust",
+    "ts",
+    "js",
+    "swift",
+    "objc",
+    "toml",
+    "py",
+    "ruby",
+    "c",
+    "cpp",
+    "go",
+    "java",
+    "csharp",
+    "kotlin",
+    "elisp",
+    "elm",
+    "docker",
+    "html",
+    "treesitterquery",
+    "json",
+    "css",
+    "lua",
+];
+
+#[inline(always)]
+pub fn get_supported_language_name_for_ignore_select(
+    supported_language: SupportedLanguage,
+) -> &'static str {
+    SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT[supported_language]
+}
+
 pub static ALL_SUPPORTED_LANGUAGES_BY_NAME_FOR_IGNORE_SELECT: Lazy<
```

Comment:
I think this is still "its own separate thing", it's sort of a "special case" I guess where we have a usage pattern where we need to do a "reverse mapping" from some "non-primary" thing back to the main "token" type